packages/runtime/src/runtimes/react/hooks/use-register-document.ts

```diff
@@ -11,12 +12,24 @@ export function useRegisterDocument(document: Document): void {
   const isInBuilder = useIsInBuilder()
   const dispatch = useDispatch()
 
-  useEffect(() => dispatch(registerDocumentsEffect([document])), [document])
+  const documentArray = useMemo(() => [document], [document])
+
+  if (isServer()) {
+    dispatch(registerDocumentsEffect(documentArray))
```

Comment:
Dispatching an effect on the server feels inappropriate to me given that we ignore the returned cleanup function